astropy/io/fits/fitstime.py

```diff
@@ -175,22 +214,71 @@ def _verify_column_info(column_info, global_info):
             column_info['format'] = global_info.get('format', None)
 
-        # Non-linear coordinate types have "4-3" form and are not time coordinates
-        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+        else:
+            # Coordinate type is either an unrecognized local time scale
+            # or a linear coordinate type
             return False
 
-        # Coordinate type is either an unrecognized local time scale
-        # or a linear coordinate type
-        return False
+    elif (unit is not None and unit in FITS_TIME_UNIT) or location is not None:
+        column_info['scale'] = global_info['scale']
 
-    if global_info['scale'] is None:
+    else:
         return False
 
-    if (unit is not None and unit in FITS_TIME_UNIT) or location is not None:
-        column_info['scale'] = global_info['scale']
-        return True
+    if unit is not None:
+        column_info['format'] = UNIT_TO_FORMAT.get(unit, None)
+    else:
+        column_info['format'] = global_info['format']
 
-    return False
+    column_info['location'] = None
+
+    if location is not None:
+
+        if location == 'TOPOCENTER':
+            column_info['location'] = global_info.get('location', None)
+            if column_info['location'] is None:
+                warnings.warn(
+                    'Time column reference position "TRPOSn" value is "TOPOCENTER". '
+                    'However, the observatory position is not properly specified. '
+                    'The FITS standard does not support this and hence reference '
+                    'position will be ignored.')
+
+    elif global_ref_pos is None or global_ref_pos == 'TOPOCENTER':
+
+        column_info['location'] = global_info.get('location', None)
+        if column_info['location'] is None:
+            warnings.warn(
+                'Time column reference position "TRPOSn" is not specified. The '
+                'default value for it is "TOPOCENTER", but due to unspecified '
+                'observatory position, reference position will be ignored.')
+
+    return True
+
+
+def _get_info_if_time_column(col, global_info):
+    """
+    Check if a column without corresponding time column keywords in the
+    FITS header represents time or not. If yes, return the time column
+    information needed for its conversion to Time. 
+    """
+    if col.name.upper() == 'TIME' and col.unit in FITS_TIME_UNIT:
+        column_info = {'scale': global_info['scale'],
+                       'format': UNIT_TO_FORMAT.get(col.unit.name, None),
+                       'location': None}
+
+        global_ref_pos = global_info.get('TREFPOS', None)
+        if global_ref_pos == 'TOPOCENTER':
+            column_info['location'] = global_info.get('location', None)
+            if column_info['location'] is None:
+                warnings.warn(
+                    'Time column reference position "TRPOSn" for column "{}" is '
```

Comment:
So, we are sure here that `TRPOS` is not there, because then the column would already have been identified as a time column?